### PR TITLE
Update to version 0.5.0

### PR DESCRIPTION
**Pull request by awvwgk**
MNT: Re-rendered with conda-build 3.21.6, conda-smithy 3.14.3, and conda-forge-pinning 2021.11.20.19.20.32

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* ~~Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by awvwgk** [comment]
@certik, you are probably more familiar with MacOS. Any idea why the new build script is failing? I had to change our build script because an update in conda-build seems to disallow to have any cyclic dependencies, even if they are just in build and no real cycles. This could also be a bug in conda-build.

I did attempt to use `FC_FOR_BUILD` to compile the bootstrap version for the build platform and than run the bootstrap to compile the actual source for the host platform. For the native build we `FC_FOR_BUILD` and `FC` are identical, but in case of cross-compilation we have different compilers in both steps. For some reason this strategy only works on Linux, but not MacOS.

**Comment by certik** [comment]
The build log has:
```
/Users/runner/miniforge3/conda-bld/fpm_1637514521157/work/conda_build.sh: line 10: /Users/runner/miniforge3/conda-bld/fpm_1637514521157/_build_env/bin/-gfortran
```
The extra `-` is probably why it fails.

Do you know the exact commands how to test this locally? We should document it in fpm's readme, how to test the conda packages locally. I am happy to execute them and debug this a bit. Although I have Apple M1, but it might give the same error.

**Comment by awvwgk** [comment]
The native MacOS/x86_64 issue is strange, looks like there is an incorrect export from the *gfortran* package:

```
+FC_FOR_BUILD=$BUILD_PREFIX/bin/-gfortran
```

This is something I could patch around, but I'm more concerned about the MacOS/arm64 failure:

```
+ /Users/runner/miniforge3/conda-bld/fpm_1637514530470/_build_env/bin/x86_64-apple-darwin13.4.0-gfortran -J build/bootstrap -o build/bootstrap/fpm fpm-0.5.0.F90
x86_64-apple-darwin13.4.0-gfortran: warning: couldn’t understand version 11.0

ld: library not found for -lgcc_s.10.4
collect2: error: ld returned 1 exit status
```

**Comment by certik** [comment]
It's probably something with the cross compilation. I tested the latest "main" (cc8a92006cdb1be73457734f354a4abff5167555) on M1 and it builds just fine (with an older `fpm` that I had installed).

**Comment by awvwgk** [comment]
So we go back to build fpm on MacOS/arm64 with the MacOS/x86_64 binary. I just have to find a way to trick conda-build into allowing us to do this ;).

**Comment by certik** [comment]
Here is how we do it with LFortran, I don't know if anything from that might help:

https://github.com/conda-forge/lfortran-feedstock/blob/7570b7a934ddfa929b872bef172f921e79d5fda4/recipe/build.sh

It runs x86 lfortran binary to cross-compile the runtime library, and then it builds the arm lfortran and copies over the runtime library.

**Comment by awvwgk** [comment]
Found a solution in the HDF5 feedstock.

https://github.com/conda-forge/hdf5-feedstock/blob/77e1a3fac883cd646419808acb5138415b1690da/recipe/build.sh#L104-L105

**Comment by certik** [comment]
Oh yes, I had so many troubles with `MACOSX_DEPLOYMENT_TARGET` before...

File: recipe/build.sh [hunk 1]
```diff
@@ -1,10 +1,26 @@
 #!/usr/bin/env bash
 set -ex
 
-export FPM_C_COMPILER="${CC}"
-if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == 1 ]]; then
-  fpm install --compiler "${FC}" --flag "${LDFLAGS} ${FFLAGS}" --prefix "${PREFIX}"
+if [[ "${CONDA_BUILD_CROSS_COMPILATION:-0}" == "1" ]]; then
+  # MACOSX_DEPLOYMENT_TARGET is for the target_platform and not for build_platform
+  unset MACOSX_DEPLOYMENT_TARGET
```
Comment (isuruf) [inline recipe/build.sh:6]: You need to set this back after the bootstrapper version has been built.

Comment (certik) [reply recipe/build.sh:6]: Thanks @isuruf! 

@awvwgk, do you have time to send it as a new PR?

**Pull request merged**